Source/WebCore/Modules/mediastream/gstreamer/GStreamerStatsCollector.cpp

```diff
@@ -40,23 +40,27 @@ namespace WebCore {
 
 RTCStatsReport::Stats::Stats(Type type, const GstStructure* structure)
     : type(type)
-    , id(String::fromLatin1(gst_structure_get_string(structure, "id")))
 {
     if (auto value = gstStructureGet<double>(structure, "timestamp"_s))
         timestamp = *value;
+
+    if (auto value = gstStructureGet<ASCIILiteral>(structure, "id"_s))
+        id = String(*value);
```

Comment:
And here you have another dangeous/wrong instance. As `*value`  is an `ASCIILiteral`,  `String()` is going to invoke `StringImpl::create` and for a `ASCIILiteral` the implementation is `ALWAYS_INLINE static Ref<StringImpl> create(ASCIILiteral literal) { return createWithoutCopying(literal.span8()); }`. This is a big life-time problem.

And from what I see below, you have lots of this below.